src/Microsoft.ML.CpuMath/SseIntrinsics.cs

```diff
@@ -755,7 +755,7 @@ public static unsafe void AddScalarU(float scalar, Span<float> dst)
             {
                 float* pDstEnd = pdst + dst.Length;
                 float* pDstCurrent = pdst;
-                int destinationEnd = pDstEnd - 4;
+                float* destinationEnd = pDstEnd - 4;
```

Comment:
I'm confused at the difference between `pDstEnd` and `destinationEnd`. They seem to mean the same thing (since they have the same words in the name), but they aren't.

Can these variables be renamed to better describe what they are?

Also (nit) I would think that the variables would have the same naming convention.  `pDstEnd` vs. `destinationEnd`.

Suggestion:  `destinationEnd` => `pVectorizationEnd`